src/hooks/useNotificationManager.tsx

```diff
@@ -0,0 +1,147 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { AppState, AppStateStatus } from 'react-native';
+import {
+  PrivatePostNotification,
+  useNotifications,
+  Notification,
+} from './useNotifications';
+import { useAsyncStorage } from './useAsyncStorage';
+import { onNotificationReceived } from '../common';
+import { compact } from 'lodash';
+
+export interface NotificationsManagerContextType {
+  unreadCount: number | undefined;
+  setNotificationsRead: () => void;
+  /**
+   * undefined - async storage has not been loaded yet
+   * null - async storage has been loaded but no value
+   * Date - async storage has loaded and key was set
+   */
+  lastReadAt: Date | null | undefined;
+  privatePostsUserIds: string[] | undefined;
+}
+
+export const NotificationsManagerContext =
+  createContext<NotificationsManagerContextType>({
+    unreadCount: 0,
+    privatePostsUserIds: [],
+    setNotificationsRead: () => {},
+    lastReadAt: null,
+  });
+
+export const NotificationsManagerProvider = ({
+  children,
+}: {
+  children?: React.ReactNode;
+}) => {
+  const [lastNotificationReadTime, setLastNotificationReadTime] =
+    useAsyncStorage('lastNotificationReadTime');
+  const { data, refetch } = useNotifications();
+
+  /**
+   * We need a ref here so we can have access to the current value inside the
+   * closure created grabbing the last read time from async storage
+   */
+  const lastReadAtRef = useRef<NotificationsManagerContextType['lastReadAt']>();
+  const [lastReadAt, setLastReadAt] =
+    useState<NotificationsManagerContextType['lastReadAt']>();
+
+  const updateLastReadAt = (readAt: Date | null) => {
+    setLastReadAt(readAt);
+    lastReadAtRef.current = readAt;
+  };
+
+  /**
+   * Load persisted lastReadAt from async storage
+   */
+  useEffect(() => {
+    if (lastNotificationReadTime.isFetchedAfterMount) {
+      updateLastReadAt(
+        lastNotificationReadTime.data
+          ? new Date(lastNotificationReadTime.data)
+          : null,
+      );
+    }
+  }, [
+    lastNotificationReadTime.data,
```

Comment:
Should `updateLastReadAt` be included here?

Reply:
The linter was even warning about it. Fixed.